gamemodes/terrortown/gamemode/client/cl_armor.lua

```diff
@@ -0,0 +1,60 @@
+ARMOR = {}
+
+-- HANDLE ARMOR STATUS ICONS
+-- init icons
+function ARMOR:Initialize()
+	STATUS:RegisterStatus("ttt_weapon_armor", {
+		hud = {
+			Material("vgui/ttt/perks/hud_armor.png"),
+			Material("vgui/ttt/perks/hud_armor_reinforced.png")
+		},
+		type = "good"
+	})
+end
+
+-- switch between icons
+local function HandleArmorStatusIcons(ply)
+	-- removed armor
+	if STATUS:Active("ttt_weapon_armor") and ply.armor == 0 then
+		STATUS:RemoveStatus("ttt_weapon_armor")
+	end
+
+	-- check if reinforced
+	local icon_id = 1
+
+	if not GetGlobalBool("ttt_armor_classic", false) then
+		icon_id = ply:ArmorIsReinforced() and 2 or 1
+	end
+
+	-- added armor
+	if not STATUS:Active("ttt_weapon_armor") and ply.armor > 0 then
+		STATUS:AddStatus("ttt_weapon_armor", icon_id)
+	end
+
+	-- normal armor level change
+	if STATUS:Active("ttt_weapon_armor") then
```

Comment:
Why this needs to be checked again?

Reply:
It is checked to make sure the icon is only changed when there should be a change. Without this, the function for the sidebar is called on every armorchange. This should be better performancewise

Reply:
better performance would be to break in the first lines if this is the intention instead of doing useless checks...

Reply:
One could add a return there, yes. But SetStatus is still heavier than two checks. Additionally you can see, that I have three scenarios, Add, Remove and Change.
One could always call Remove and Add, but this is bad in my opinion. Therefore these checks, basically as "edge detection" of a signal change.